space-concordia-robotics/robotics-prototype
# Docker configuration 418

# Assignee Section

## Description
This PR dockerizes the project, allowing users to run it regardless of their environment.
Using docker-compose, three services (group of containers based on the same docker image) are created:
- One running roscore called `ros-master`
- One running rosbridge called `rosbridge`
- One running the actual project called `prototype`

### Steps for Testing

1. Follow the `README` instructions in the `docker` directory to run the project.
2. Once everything is up and running, make sure you see the GUI at http://localhost:5000
3. Open a terminal and run `docker exec -it docker_rosbridge_1 bash`. This gives you a shell inside the container running rosbridge
4. Still inside the container, run `rostopic pub -1 /battery_voltage mcu_control/Voltage 15.0`
5. You should see the voltage displayed on the GUI

closes #418 (doesn't include the travis part)

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
